flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnResourceManagerFactory.java

```diff
@@ -81,4 +91,27 @@ public static YarnResourceManagerFactory getInstance() {
 			webInterfaceUrl,
 			resourceManagerMetricGroup);
 	}
+
+	@Override
+	protected CPUResource getDefaultCpus(final Configuration configuration) {
+		int fallback = configuration.getInteger(YarnConfigOptions.VCORES);
+		double cpuCoresDouble = TaskExecutorProcessUtils.getCpuCoresWithFallback(configuration, fallback).getValue().doubleValue();
+		@SuppressWarnings("NumericCastThatLosesPrecision")
+		long cpuCoresLong = Math.max((long) Math.ceil(cpuCoresDouble), 1L);
+		//noinspection FloatingPointEquality
+		if (cpuCoresLong != cpuCoresDouble) {
```

Comment:
True, we have not exposed `taskmanager.cpu.cores`. But we might in the future, and if we do, it could be easily overlooked to add this check later. Again, I believe there's no harm for over checking it.